components/camel-jms/src/main/java/org/apache/camel/component/jms/StreamMessageInputStream.java

```diff
@@ -48,6 +48,11 @@ public int read() throws IOException {
     public int read(byte[] array) throws IOException {
         try {
             int num = message.readBytes(array);
+            if (num < 0) {
+                //the first 128K(FileUtil.BUFFER_SIZE/128K is used when sending JMS StreamMessage) 
+                //buffer reached, give a chance to see if there is the next 128K buffer
+                num = message.readBytes(array);
+            }
```

Comment:
If the payload > 2*128K, this can only read first two chuncks ?

Reply:
I think it need to remove ```eof = num < 0```.

Reply:
Hi @zhfeng,

Thanks for your feedback!

The comment may be misleading, but the logic here actually will read all n*128k chunk until the end of the JMS StreamMessage.  The point is that when finishing reading 1 chunk, give chance to read the next chunk, and don't return a negative num, and all caller code which call into StreamMessageInputStream(which is an InputStream) read(byte[] array) will keep the read operation going.

Cheers
Freeman

Reply:
Well, how can we make sure there is the last chunk ? From the java docs https://docs.oracle.com/javaee/7/api/javax/jms/StreamMessage.html#readBytes-byte:A-
```
To read the field value, readBytes should be successively called until it returns a value less than the length of the read buffer. The value of the bytes in the buffer following the last byte read is undefined.
```

Reply:
I suggest to remove the next line
```
eof = num  < 0;
```
since we can not to see the eof if the num is -1 because there will be more chunks.

Reply:
And there might be an indicator to mark the last chunk ?

Reply:
Ok, I understand now. The last readBytes return -1 means that we have no next chunk to read. Your fix is fine.

Reply:
Yep, ;) 
Thanks @zhfeng !